File: test_nguonc.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  final url = 'https://phim.nguonc.com/api/films/tim-kiem?keyword=thu+ky+kim+sao+the&limit=1';
  final res = await http.get(Uri.parse(url));
  final data = json.decode(res.body);
  print('nguonc type: ${data['items'][0]['type'] ?? data['items'][0]['type_name'] ?? 'none'}');
}

File: test_nguonc.py
import urllib.request
import json
import sys

sys.stdout.reconfigure(encoding='utf-8')
url = "https://phim.nguonc.com/api/film/su-huynh-qua-can-trong"
try:
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    res = urllib.request.urlopen(req).read().decode('utf-8')
    data = json.loads(res)
    item = data.get('movie', {})
    print("Name:", item.get('name'))
    print("Original Name:", item.get('original_name'))
except Exception as e:
    print(e)
